### PR TITLE
Clock reduction tests

### PR DESCRIPTION
**Pull request by Laursen79**

### CODE REVIEW

File: samples/json/ClockReductionTest/ReplacedClocks/Components/Clock replaced simpel.json [hunk 1]
```diff
@@ -0,0 +1,113 @@
+{
```
Comment (alexsteffensen) [inline samples/json/ClockReductionTest/ReplacedClocks/Components/Clock replaced simpel.json:1]: Consider changing the file name

File: src/tests/ClockReduction/unused_clock_detection_test.rs [hunk 2]
```diff
@@ -0,0 +1,40 @@
+#[cfg(test)]
+mod unused_clocks_tests {
+    use crate::tests::ClockReduction::helper::test::assert_duplicated_clock_detection;
+    use crate::DataReader::json_reader::read_json_component;
+    use std::collections::HashSet;
+
+    //TODO: Merge these functions
```
Comment (t-lohse) [inline src/tests/ClockReduction/unused_clock_detection_test.rs:7]: todo?

File: src/ModelObjects/component.rs [hunk 3]
```diff
@@ -255,6 +255,218 @@ impl Component {
...
+                ClockReductionReason::Duplicate(global) => self.replace_clock(&clock, global),
+                ClockReductionReason::Unused => self.remove_clock(&clock.updates),
+            }
+
+            let clock_val = *self.declarations.clocks.get(clock.clock.as_str()).unwrap();
+            self.declarations.clocks.values_mut().for_each(|val| {
+                if *val > clock_val {
+                    *val -= 1;
+                }
+            });
```
Comment (t-lohse) [inline src/ModelObjects/component.rs:274]:
```rust
self.declarations.clocks.values_mut().filter(|val| *val > clock_val).for_each(|val| *val -= 1);
```

File: samples/json/EcdarUniversity/Components/Machine.json [hunk 4]
```diff
@@ -173,4 +173,4 @@
   "height": 390.0,
   "color": "7",
   "includeInPeriodicCheck": false
-}
\ No newline at end of file
+}
```
Comment (t-lohse) [inline samples/json/EcdarUniversity/Components/Machine.json:176]: There shouldn't be a change here. Please reset to `main` branch

File: src/tests/ClockReduction/unused_clock_detection_test.rs [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+            "samples/json/ClockReductionTest/UnusedClockWithCycle",
+            component_name,
+        );
+
+        let unused_clocks = component.find_redundant_clocks();
+
+        assert_duplicated_clock_detection(&unused_clocks, 1, HashSet::from([unused_clock]), true);
+    }
+
+    //TODO: Merge these functions
```
Comment (t-lohse) [inline src/tests/ClockReduction/unused_clock_detection_test.rs:19]: todo?

**Review by t-lohse — CHANGES_REQUESTED** [review]
comments

File: samples/json/ClockReductionTest/ReplacedClocks/Components/Clock replaced.json [hunk 6]
```diff
@@ -0,0 +1,99 @@
+{
```
Comment (alexsteffensen) [inline samples/json/ClockReductionTest/ReplacedClocks/Components/Clock replaced.json:1]: Please, change file name

**Review by t-lohse — APPROVED** [review]

**Pull request merged**